lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -819,6 +819,100 @@ spec = do
         expectResponseCode @IO HTTP.status404 r
         expectErrorMessage (errMsg404NoWallet $ w ^. walletId) r
 
+    it "TRANS_ESTIMATE_08 - Fee estimation of edge cases" $ \ctx -> do
+        let walletBalance = 5000000
+        wSrc <- fixtureWalletWith @n ctx [walletBalance]
+        wDest <- emptyWallet ctx
+        addrs <- listAddresses @n ctx wDest
+        let destination = (addrs !! 1) ^. #id
```

Comment:
I've been fighting this "anti-pattern" from our integration DSL recently :grimacing: 
Restoring a wallet simply to get addresses that can be obtained offline is rather counter-productive. Instead, use:

- https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core-integration/src/Test/Integration/Framework/DSL.hs#L1183
- https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core-integration/src/Test/Integration/Framework/DSL.hs#L1208
- `mw <- entropyToMnemonic <$> genEntropy` to get a mnemonic sentence.

Reply:
On that one, I had a problem some time ago with this way of generating shelley addresses actually. Shelley addresses (generated with https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core-integration/src/Test/Integration/Framework/DSL.hs#L1208) were actually failing in Jormungandr tests giving `400` -> see Comment on this PR https://github.com/input-output-hk/cardano-wallet/pull/1711. We were not able to find out what's the issue actually (with @paweljakubas), so went for the "anti-pattern" in the end as it was working...

Reply:
Shelley addresses are different from jormungandr addresses. We have 4 types of addresses: Byron, icarus, jormungandr and Shelley.